aiida_quantumespresso/calculations/pw.py

```diff
@@ -160,17 +160,14 @@ def validate_inputs(value, _):
         """Validate the top level namespace.
 
         1. Check that the restart input parameters are set correctly. In case of 'nscf' and 'bands' calculations, this
-        means that ``parent_folder`` is provided, ``startingpot`` is set to 'file' and ``restart_mode`` is
-        'from_scratch'. For other calculations, if the ``parent_folder`` is provided, the restart settings must be set
-        to use some of the outputs.
+        means that ``startingpot`` is set to 'file' and ``restart_mode`` is 'from_scratch'. For other calculations, if
+        the ``parent_folder`` is provided, the restart settings must be set to use some of the outputs.
         """
         parameters = value['parameters'].get_dict()
         calculation_type = parameters.get('CONTROL', {}).get('calculation', 'scf')
 
         # Check that the restart input parameters are set correctly
         if calculation_type in ('nscf', 'bands'):
-            if 'parent_folder' not in value:
-                return f'`parent_folder` not provided for `{calculation_type}` calculation.'
```

Comment:
where is this going to get checked now? Do we have a check somewhere in the outline steps or is it just going to horribly except when it tries to call `self.inputs.parent_folder` somewhere? If that is the case, it should be caught and transformed in an understandable exception message. I also think we should maybe open an issue describing this use case to see if we can come up with a solution that allows to still have the check, but have it disabled in wrapping work chains that guarantee that they will be specifying it themselves.

Reply:
I'd have to double check, but I think that for both the `PwBandsWorkChain` and `PdosWorkChain`, this validation is unnecessary. The PwBandsWorkChain _always_ runs an SCF first, and the `PdosWorkChain` has its own validation for the `parent_folder` input (i.e. either an SCF is run, or the `parent_folder` must be provided for the `nscf`. So the only issue is that for a bare `nscf`/`bands` `PwCalculation`, or one wrapped in the `PwBaseWorkChain`, the QE will fail since it doesn't have the required files in the folder.

I suppose we could add an `exclude_validation` argument to the `expose_inputs` method (similar to how we use `exclude` to exclude inputs). The user could pass a tuple that is very similar to the `exclude` one (i.e. using the `.` notation for inputs nested in `PortNameSpaces`. One question would be how to indicate _all_ validation, or only the top-level validation should be skipped. Let's indeed open an issue to discuss this.

For this PR: I think that we can simply remove this validation for now, and say that validating that an `nscf` or `bands` calculation has the `parent_folder` input is the responsibility of the work chains that wrap them, as it was before #722.